src/utils/mod.rs

```diff
@@ -1674,31 +1742,58 @@ pub enum PublicKey {
 
 /// Struct holding a pcr value.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct PcrValue {
+pub struct Digest {
     value: Vec<u8>,
 }
 
-impl PcrValue {
+impl Deref for Digest {
+    type Target = Vec<u8>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.value
+    }
+}
+
+impl Digest {
     /// Function for retrieving the value.
     pub fn value(&self) -> &[u8] {
         &self.value
     }
 }
 
-impl TryFrom<TPM2B_DIGEST> for PcrValue {
+impl TryFrom<TPM2B_DIGEST> for Digest {
     type Error = Error;
     fn try_from(tss_digest: TPM2B_DIGEST) -> Result<Self> {
         let size = tss_digest.size as usize;
         if size > 64 {
             error!("Error: Invalid TPM2B_DIGEST size(> 64)");
             return Err(Error::local_error(WrapperErrorKind::InvalidParam));
         }
-        Ok(PcrValue {
+        Ok(Digest {
             value: tss_digest.buffer[..size].to_vec(),
         })
     }
 }
 
+impl TryFrom<Digest> for TPM2B_DIGEST {
+    type Error = Error;
+    fn try_from(digest: Digest) -> Result<Self> {
+        if digest.len() > 64 {
+            return Err(Error::local_error(WrapperErrorKind::WrongParamSize));
+        }
+        let mut buffer: [u8; 64] = [0; 64];
+        for (pos, val) in digest.iter().enumerate() {
+            buffer[pos] = *val;
+        }
+        Ok(TPM2B_DIGEST {
+            size: digest.len() as u16,
+            buffer,
+        })
+    }
+}
+
+type PcrValue = Digest;
```

Comment:
Apart from slightly enhanced reading, is this actually needed? I don't think the type is actually referred to outside this file.

Reply:
Given that a `PcrValue` is not exactly the same use case, I personally like to have this aliased name, just for readability.
If you disagree, I'm happy to change it.

Reply:
Happy to keep it!